nocodb/Base.py

```diff
@@ -61,8 +61,10 @@ def get_table_by_title(self, title: str) -> Table:
             raise Exception(f"Table with name {title} not found!")
 
     def create_table(self, table_name: str,
-                      columns: list[dict] = [], add_default_columns: bool = True, 
+                      columns: list[dict] = None, add_default_columns: bool = True,
                      **kwargs) -> Table:
+        if columns is None:
+            columns = []
```

Comment:
What is the point of having `None` the default value? And you also have to update the type hint as well.

```suggestion
                      columns: list[dict]|None = None, add_default_columns: bool = True,
                     **kwargs) -> Table:
        if columns is None:
            columns = []
```

Same in the other file. Is it some automatic linting from your editor?

Reply:
> What is the point of having `None` as the default value?

This technique is used to avoid a "mutable defaults" problem. This point clearly explained in [StackOverflow thread](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument) and in many tutorials [like this](https://docs.python-guide.org/writing/gotchas/). In most cases such behaviour must be avoided, excepting cases that you exactly know what you do.

> you have to update the type hint as well

IMO it is not necessary in this case, that notation used just for type hinting, and actually you not expect `None` to be passed as argument from user code, because `None` is a default value.

> Is it some automatic linting from your editor?

Partly, my editor (PyCharm) *suggests* to correct this, but I work with Python every day for couple of years and pay attention on snippets like this in code reviews for colleagues.

Reply:
VSCode freaks out from this, so we need that `|None`:
![kép](https://github.com/user-attachments/assets/b6c49f4b-1e92-4369-9a2d-ac8194184c43)

Also 2 lines lower there is an `if not columns`, we don't need the `is None`:

```suggestion
                      columns: list[dict] | None = None, add_default_columns: bool = True,
                     **kwargs) -> Table:
          kwargs["table_name"] = table_name
        if not columns:
          kwargs["columns"] = Column.get_id_metadata()
```